src/axom/core/Macros.hpp

```diff
@@ -35,6 +35,22 @@
 #define AXOM_HOST
 #endif
 
+/*
+ * \def AXOM_SUPPRESS_HD_WARN
+ *
+ * \brief Macro used to silence __host__ __device__ compiler warnings
+ *  when calling __host__ function from __host__ __device function.
+ */
+#if defined(__CUDACC__)
+  #ifdef WIN32
+  #define AXOM_SUPPRESS_HD_WARN __pragma(nv_exec_check_disable)
```

Comment:
I suggest we split this into two separate macros: (a) `AXOM_PRAGMA` and (b) `AXOM_SUPPRESS_HD_WARN`. That way, we can reuse the `AXOM_PRAGMA` in other places as needed.

Something like this:
```
#ifdef WIN32
#define AXOM_PRAGMA(__supplied_pragma) __pragma(__supplied_pragma)
#else 
#define AXOM_PRAGMA(__supplied_pragma) _Pragma("__supplied_pragma")
#endif

#if defined(__CUDACC__)
#define AXOM_SUPPRESS_HD_WARN AXOM_PRAGMA(nv_exec_check_disable)
#else 
#define AXOM_SUPPRESS_HD_WARN AXOM_PRAGMA
#endif 
```

**Note:** There is a way to handle the `"` inside a macro, I think you just add a`#` in front of it or something along those lines